lib/internal/modules/esm/hooks.js

```diff
@@ -126,16 +125,18 @@ class Hooks {
    */
   async register(urlOrSpecifier, parentURL) {
     const moduleLoader = require('internal/process/esm_loader').esmLoader;
-
     const keyedExports = await moduleLoader.import(
       urlOrSpecifier,
       parentURL,
       kEmptyObject,
     );
-
     this.addCustomLoader(urlOrSpecifier, keyedExports);
   }
 
+  allowImportMetaResolve() {
+    return false;
+  }
```

Comment:
I imagined the following:

```ts
interface ModuleLoadyThing {
  allowImportMetaResolve(): boolean;
  load(url: string, context: object): Promise<...>
  resolve(originalSpecifier: string, parentURL: string, importAssertions: string): Promise<...>
  register(specifier: string, parentUrl: string) { ... }
}

class ModuleLoader implements ModuleLoadyThing { ... }
class Hooks implements ModuleLoadyThing { ... }
```

And because the root module loader is the only thing that's ever accessed, `allowImportMetaResolve` invokes the delegate's `allowImportMetaResolve` (in this case `Hooks.allowImportMetaResolve`). I guess I could make it a read-only property and copy the value over on calls to `setDelegate` if that's preferable?

Reply:
This interface is now updated to add `resolveSync`:

```ts
interface ModuleLoadyThing {
  allowImportMetaResolve(): boolean;
  load(url: string, context: object): Promise<...>
  resolve(originalSpecifier: string, parentURL: string, importAssertions: string): Promise<...>
  resolveSync(originalSpecifier: string, parentURL: string, importAssertions: string): ...
  register(specifier: string, parentUrl: string) { ... }
}
```

Reply:
This interface is now updated to be:

```ts
interface ModuleLoadyThing {
  allowImportMetaResolve: boolean;
  load(url: string, context: object): Promise<...>
  resolve(originalSpecifier: string, parentURL: string, importAssertions: string): Promise<...>
  resolveSync(originalSpecifier: string, parentURL: string, importAssertions: string): ...
  register(specifier: string, parentUrl: string): unknown
}
```